python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -1,30 +1,135 @@
 # Copyright (c) 2023-2025, NVIDIA CORPORATION.
 
 from cython.operator cimport dereference
-from libc.stdint cimport uintptr_t
+
+from cpython.pycapsule cimport (
+    PyCapsule_GetPointer,
+    PyCapsule_New,
+)
+
 from libcpp.limits cimport numeric_limits
 from libcpp.memory cimport make_unique, unique_ptr
 from libcpp.utility cimport move
-from rmm.pylibrmm.stream cimport Stream
+
 from pylibcudf.libcudf.column.column cimport column, column_contents
 from pylibcudf.libcudf.column.column_factories cimport make_column_from_scalar
-from pylibcudf.libcudf.scalar.scalar cimport scalar
-from pylibcudf.libcudf.types cimport size_type
+from pylibcudf.libcudf.interop cimport ArrowArray, ArrowSchema, arrow_column
+from pylibcudf.libcudf.scalar.scalar cimport scalar, numeric_scalar
+from pylibcudf.libcudf.types cimport size_type, size_of as cpp_size_of
+from pylibcudf.libcudf.utilities.traits cimport is_fixed_width, is_fixed_point
+from pylibcudf.libcudf.copying cimport get_element
+
+from pylibcudf.libcudf.interop cimport (
+    ArrowArray,
+    ArrowSchema,
+    column_metadata,
+    to_arrow_host_raw,
+    to_arrow_schema_raw,
+)
 
 from rmm.pylibrmm.device_buffer cimport DeviceBuffer
+from rmm.pylibrmm.stream cimport Stream
 
 from .gpumemoryview cimport gpumemoryview
 from .filling cimport sequence
 from .scalar cimport Scalar
 from .types cimport DataType, size_of, type_id
 from .utils cimport int_to_bitmask_ptr, int_to_void_ptr, _get_stream
+from .null_mask cimport bitmask_allocation_size_bytes
+from ._interop_helpers cimport (
+    _release_schema,
+    _release_array,
+    _metadata_to_libcudf,
+)
+from ._interop_helpers import ColumnMetadata
 
-from functools import cache
+import functools
 
 
 __all__ = ["Column", "ListColumnView", "is_c_contiguous"]
 
 
+class _ArrowLikeMeta(type):
+    def __subclasscheck__(cls, other):
+        return hasattr(other, "__arrow_c_array__")
+
+
+class _ArrowLike(metaclass=_ArrowLikeMeta):
+    pass
+
+
+cdef class _ArrowColumnHolder:
+    """A holder for an Arrow column for gpumemoryview lifetime management."""
+    cdef unique_ptr[arrow_column] col
+
+
+cdef class OwnerWithCAI:
+    """An interface for column view's data with gpumemoryview via CAI."""
+    @staticmethod
+    cdef create(column_view cv, object owner):
+        obj = OwnerWithCAI()
+        obj.owner = owner
+        cdef int size
+        cdef column_view offsets_column
+        cdef unique_ptr[scalar] last_offset
+        if cv.type().id() == type_id.EMPTY:
+            size = cv.size()
+        elif is_fixed_width(cv.type()) or is_fixed_point(cv.type()):
```

Comment:
```suggestion
        elif is_fixed_width(cv.type()):
```

Reply:
Good catch. Done in https://github.com/rapidsai/cudf/pull/18402/commits/457f67742385d80a275fa649196678ff4374aaab